worlds/pokemon_rb/locations.py

```diff
@@ -5,46 +5,48 @@
 loc_id_start = 172000000
 
 
-def trainersanity(multiworld, player):
-    return multiworld.trainersanity[player]
+def trainersanity(world, player):
+    include = world.trainersanity_table.pop(0)
+    world.trainersanity_table.append(include)
+    return include
 
 
-def dexsanity(multiworld, player):
-    include = multiworld.worlds[player].dexsanity_table.pop(0)
-    multiworld.worlds[player].dexsanity_table.append(include)
+def dexsanity(world, player):
+    include = world.dexsanity_table.pop(0)
+    world.dexsanity_table.append(include)
     return include
 
 
-def hidden_items(multiworld, player):
-    return multiworld.randomize_hidden_items[player]
+def hidden_items(world, player):
+    return world.options.randomize_hidden_items
 
 
-def hidden_moon_stones(multiworld, player):
-    return multiworld.randomize_hidden_items[player] or multiworld.stonesanity[player]
+def hidden_moon_stones(world, player):
+    return world.options.randomize_hidden_items or world.options.stonesanity
 
 
-def tea(multiworld, player):
-    return multiworld.tea[player]
+def tea(world, player):
+    return world.options.tea
 
 
-def extra_key_items(multiworld, player):
-    return multiworld.extra_key_items[player]
+def extra_key_items(world, player):
+    return world.options.extra_key_items
 
 
-def always_on(multiworld, player):
+def always_on(world, player):
     return True
 
 
-def prizesanity(multiworld, player):
-    return multiworld.prizesanity[player]
+def prizesanity(world, player):
+    return world.options.prizesanity
 
 
-def split_card_key(multiworld, player):
-    return multiworld.split_card_key[player].value > 0
+def split_card_key(world, player):
+    return world.options.split_card_key.value > 0
 
 
-def not_stonesanity(multiworld, player):
-    return not multiworld.stonesanity[player]
+def not_stonesanity(world, player):
+    return not world.options.stonesanity
```

Comment:
Couple of things I don't love about this

1. The "player" attribute is now unused in all of these and could be deleted. Obviously that's some amount of work since every call has to be adjusted too, but I think it's pertinent.
2. With the new options system, having these functions seems. Kinda slow and unnecessary? Is it really worth that much to have these functions instead of just inlining `world.options.stonesanity` in the access rule itself?

**Now**, I will say that both of these things can be considered "out of scope" for this PR. What you've got here is an objective improvement, so I won't block over this.
However, I'd encourage having another look at this in a future PR.